.github/github_workflow_scripts/purge_branch_protection_rules.py

```diff
@@ -0,0 +1,343 @@
+"""
+A CLI to manage GitHub branch protection rules.
+
+It uses GitHub GraphQL API to query and remove
+branch protection rules.
+"""
+
+from dataclasses import dataclass
+import os
+from pathlib import Path
+import sys
+from typing import Any
+
+import tabulate
+from utils import get_logger
+
+from github import Github
+import github
+import github.Requester
+
+
+DEFAULT_REPO = "demisto/content"
+
+GH_TOKEN_ENV_VAR = "GITHUB_TOKEN"
+
+# https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+# e.g. 'demisto/content'
+GH_REPO_ENV_VAR = "GITHUB_REPOSITORY"
+GH_JOB_SUMMARY_ENV_VAR = "GITHUB_STEP_SUMMARY"
+PROTECTED_RULES = ["contrib/**/*", "master"]
+
+logger = get_logger(f"{Path(__file__).stem}")
+
+
+@dataclass
+class BranchProtectionRule:
+    id: str
+    pattern: str
+    matching_refs: int
+
+
+# Queries
+GET_BRANCH_PROTECTION_GRAPHQL_QUERY_TEMPLATE = """query($owner: String!, $name: String!) {
+repository(owner: $owner, name: $name) {
+    branchProtectionRules(first: 100) {
+    nodes {
+        id
+        pattern
+        matchingRefs(first: 10) {
+        totalCount
+        }
+    }
+    }
+}
+}"""
+
+DELETE_BRANCH_PROTECTION_RULE_QUERY_TEMPLATE = """mutation deleteBranchProtectionRule($branchProtectionRuleId: ID!) {
+    deleteBranchProtectionRule(input: {branchProtectionRuleId: $branchProtectionRuleId}) {
+        clientMutationId
+    }
+}"""
+
+
+# Helper Functions
+
+def get_repo_owner_and_name() -> tuple[str, str]:
+    """
+    Extracts the repository owner and name from a given repository string.
+
+    Returns:
+        A `Tuple[str, str]` containing the repository owner and name.
+
+    Raises:
+        `ValueError`: If the input string is not in the expected 'owner/repository' format.
+    """
+
+    repo = os.getenv(GH_REPO_ENV_VAR, DEFAULT_REPO)
```

Comment:
@ShahafBenYakir @kobymeir - Since this script will be available in forks for this repo, maybe it's better to remove a `DEFAULT_REPO` (value `demisto/content`) and just exit if this env var is not set?

Reply:
Where do you set it ?

Reply:
a bit earlier [in line 22](https://github.com/demisto/content/pull/35604/files/fa2b6ca0faadc6d386fd13e059767cf73a2bf464#diff-03d502a34afbd79f756537127b1365d915fece39a6a37f5ab28ac04089b8d23cR22).

Reply:
I meant where do we set the environment variable GH_REPO_ENV_VAR

Reply:
It's a [default env var](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/variables#default-environment-variables) that is set to the name of the repo in GitHub workflow:

<img width="947" alt="image" src="https://github.com/user-attachments/assets/260b0136-6c3c-4d2d-8ba4-0d2817a64801">